test/rc4test.c

```diff
@@ -61,115 +54,74 @@ static unsigned char output[7][30] = {
-        if (memcmp(obuf, output[3], data_len[3] + 1) != 0) {
-            printf("error in RC4 multi-call processing\n");
-            printf("output:");
-            for (j = 0; j < data_len[3] + 1; j++)
-                printf(" %02x", obuf[j]);
-            printf("\n");
-            printf("expect:");
-            p = &(output[3][0]);
-            for (j = 0; j < data_len[3] + 1; j++)
-                printf(" %02x", *(p++));
-            err++;
-        } else {
-            printf(".");
-            fflush(stdout);
-        }
-    }
-    printf("done\n");
-    printf("bulk test ");
-    {
-        unsigned char buf[513];
-        SHA_CTX c;
-        unsigned char md[SHA_DIGEST_LENGTH];
-        static unsigned char expected[] = {
-            0xa4, 0x7b, 0xcc, 0x00, 0x3d, 0xd0, 0xbd, 0xe1, 0xac, 0x5f,
-            0x12, 0x1e, 0x45, 0xbc, 0xfb, 0x1a, 0xa1, 0xf2, 0x7f, 0xc5
-        };
-
-        RC4_set_key(&key, keys[0][0], &(keys[3][1]));
-        memset(buf, 0, sizeof(buf));
-        SHA1_Init(&c);
-        for (i = 0; i < 2571; i++) {
-            RC4(&key, sizeof(buf), buf, buf);
-            SHA1_Update(&c, buf, sizeof(buf));
-        }
-        SHA1_Final(md, &c);
-
-        if (memcmp(md, expected, sizeof(md))) {
-            printf("error in RC4 bulk test\n");
-            printf("output:");
-            for (j = 0; j < (int)sizeof(md); j++)
-                printf(" %02x", md[j]);
-            printf("\n");
-            printf("expect:");
-            for (j = 0; j < (int)sizeof(md); j++)
-                printf(" %02x", expected[j]);
-            printf("\n");
-            err++;
-        } else
-            printf("ok\n");
+    RC4_set_key(&key, keys[3][0], &(keys[3][1]));
+    memset(obuf, 0, sizeof(obuf));
+    RC4(&key, i, &(data[3][0]), obuf);
+    RC4(&key, data_len[3] - i, &(data[3][i]), &(obuf[i]));
+    return TEST_mem_eq(obuf, data_len[3] + 1, output[3], data_len[3] + 1);
+}
+
+static int test_rc4_multi_call(const int i)
+{
+    unsigned char obuf[512];
+    RC4_KEY key;
+
+    RC4_set_key(&key, keys[3][0], &(keys[3][1]));
+    memset(obuf, 0, sizeof(obuf));
+    RC4(&key, i, &(data[3][0]), obuf);
+    RC4(&key, data_len[3] - i, &(data[3][i]), &(obuf[i]));
+    return TEST_mem_eq(obuf, data_len[3] + 1, output[3], data_len[3] + 1);
+}
+
+static int test_rc_bulk(void)
+{
+    RC4_KEY key;
+    unsigned char buf[513];
+    SHA_CTX c;
+    unsigned char md[SHA_DIGEST_LENGTH];
+    int i;
+    static unsigned char expected[] = {
+        0xa4, 0x7b, 0xcc, 0x00, 0x3d, 0xd0, 0xbd, 0xe1, 0xac, 0x5f,
+        0x12, 0x1e, 0x45, 0xbc, 0xfb, 0x1a, 0xa1, 0xf2, 0x7f, 0xc5
+    };
+
+    RC4_set_key(&key, keys[0][0], &(keys[3][1]));
+    memset(buf, 0, sizeof(buf));
+    SHA1_Init(&c);
+    for (i = 0; i < 2571; i++) {
+        RC4(&key, sizeof(buf), buf, buf);
+        SHA1_Update(&c, buf, sizeof(buf));
     }
-    EXIT(err);
+    SHA1_Final(md, &c);
+
+    return TEST_mem_eq(md, sizeof(md), expected, sizeof(md));
```

Comment:
probably really should be `TEST_mem_eq(md, sizeof(md), expected, sizeof(expected));`